session.go

```diff
@@ -157,6 +172,7 @@ func (s *session) recvLoop() {
 		if err != nil {
 			if err == io.EOF {
 				s.onSessionError(err, nil)
+				stoppedOnEOF = true
```

Comment:
Shouldn't this also be set in other EOF cases below? If so, maybe refactor into a helper method?

Reply:
The other cases that would stop this loop are not plain EOF. They could be UnexpectedEOF but those indicate unusual conditions.

Reply:
Sure but they could be plain EOF as well right?

Reply:
so that's why I was suggesting a helper method that would include the `err == io.EOF` line.

Reply:
> Sure but they could be plain EOF as well right?

No, that was my point, this is the only place where an EOF is expected. If we encounter an EOF reading any other portion of the frame, that's by definition an unexpected EOF because a client will never actually close the connection in the middle of sending a frame.